src/core/common/tasklet.hpp

```diff
@@ -96,6 +96,44 @@ class Tasklet : public InstanceLocator, public OwnerLocator
     Tasklet *mNext;
 };
 
+/**
+ * This class defines a tasklet which also tracks an arbitrary context information.
+ *
+ * In typical `Tasklet` use, in the handler callback, the owner of the tasklet is determined using `GetOwner<Type>`
+ * method. This method works if there is a single instance of `Type` within OpenThread instance hierarchy. The
+ * `ContextTasklet` is intended for cases where there may be multiple instances of the same class/type using a `Tasklet`
+ * object. `ContextTaskelt` will store a context `void *` information.
+ *
+ */
+class ContextTasklet : public Tasklet
```

Comment:
nit: what about renaming to `TaskletContext`, since this is a sub-class of `Tasklet`? Similar to `Coap` and `CoapSecure`?

Reply:
Thanks. Changed them all in new pushed commits. Changed the commit logs as well. :)